### DISCUSSION TITLE
Array in one line

### DISCUSSION THREAD

**Discussion by igor-lemon** [eemeli/yaml#220] [Q&A] [answered]
Hi! Sometimes I need to set array in one line
```json
{
  "yaml": [
    "here", 
    "some", 
    "command"
  ]
}
```

And I want to have 

```yaml
yaml: ['here', 'some', 'command']
```

But I have this result now:

```yaml
yaml:
  - here
  - some
  - command
```

**Comment by eemeli** [comment, marked as answer]
To do that you'll need to first create a `YAML.Document` instance from your object, and then set the `type` of the collection to be `'FLOW_SEQ'`. The API for doing that should get easier once #190 is resolved.

**Reply by oscard0m** [reply]
Hi @eemeli,

Could you draft a small snippet of code on how to tackle this approach? 

I would be happy to open a PR to document this if you want to.

**Reply by eemeli** [reply]
As mentioned in the parent, the API in `yaml@2` is now a bit simpler:

```js
const doc = new YAML.Document({ yaml: ['here', 'some', 'command'] })
doc.get('yaml').flow = true
doc.toString()
> 'yaml: [ here, some, command ]\n'
```

If you think there's a part of the docs that could be improved to make this easier to understand, please do file a PR.

**Reply by oscard0m** [reply]
Thanks for your help with this @eemeli , you are being very helpful.

---

Since I plan to write a script to modify YAML files adding a new option for a concrete field, and open a PR for that, my code should not be aware of the shape of the incoming YAML file structure. Also, it would be nice to preserve the original shape of the file so there is no noise introduced when opening a PR with the modifications applied to the file.

Here is an example of the issue I'm experimenting: https://github.com/oscard0m/octoherd-script-replace-all-repository-topics/pull/11/files

From what I infer from your snippet you shared, the steps I should follow for my use-case are:
- Before creating `YAML.Document` with the content of the incoming file, to be aware and mark what was the original format for each `array`/`sequence`.
- For each marked array, to enable the `flow` flag for those paths in the `YAML.Document`.

---

Would have sense to create a new `Option` in the `parse` step on this library to track each sequence/array and preserve its shape when dumping back as string? Or maybe there is a way to do it already which I'm not finding?

Thanks once again for your support.

**Reply by eemeli** [reply]
If you're modifying an existing YAML file, you should not use `parse` at all, but `parseDocument`. That gives you a Document that already holds most of the metadata associated with the structure, such as comments and the block/flow mode of collections.

**Reply by oscard0m** [reply]
> If you're modifying an existing YAML file, you should not use `parse` at all, but `parseDocument`. That gives you a Document that already holds most of the metadata associated with the structure, such as comments and the block/flow mode of collections.

With this method I think I will get to the solution I'm looking for. THANK YOU SO MUCH for assisting me on this @eemeli 🙏🏽 🙏🏽 🙏🏽 🙏🏽 🙏🏽 🙏🏽

**Reply by oscard0m** [reply]
[It worked](https://github.com/oscard0m/octoherd-script-replace-all-repository-topics/pull/19/files) @eemeli ! Thank you once again for your help 👏🏽 👏🏽 👏🏽 👏🏽

**Reply by alexmuch** [reply]
> ```js
> const doc = new YAML.Document({ yaml: ['here', 'some', 'command'] })
> doc.get('yaml').flow = true
> doc.toString()
> > 'yaml: [ here, some, command ]\n'
> ```

Is this code snippet still expected to work? When I use it, I hit a `TypeError: Cannot set property 'flow' of undefined`

I'm asking because I've tried using the flow option `{flow: true}` with a stringify and manually setting flow to true in my collection but I'm still getting the block output no matter what I do.

Edit: I made a stackblitz here showing what I mean. https://stackblitz.com/edit/node-ggxngj?file=index.js
It's simply reading yaml and writing it to the exported file. Run it with `node index.js`. Neither the property nor the option are allowing me to get the `alias` output to be in flow mode.

**Reply by eemeli** [reply]
@alexmuch The `flow` property was added to the in-progress `yaml@2` branch, and your example is using `yaml@1`.

**Reply by alexmuch** [reply]
How do I get the `yaml@2` branch? I ran `npm install yaml` and it gave me this version. Is the `yaml@2` not official yet?

**Reply by eemeli** [reply]
The installation instructions are available in the README of this repo, and at the top of the documentation site: https://eemeli.org/yaml

**Comment by igor-lemon** [comment]
What about a feature for adding this option on the fly in `addIn`, `setIn`, methods?
I work with Docker Compose configs and we use mixes with different types. Some array options via 
```yaml
[foo, bar]
```
some
```yaml
- foo
- bar
```

**Reply by oscard0m** [reply]
To preserve the sequence/array format from the original would be awesome

**Reply by eemeli** [reply]
@oscard0m That's already done by the Document layer of the API:

```js
YAML.parseDocument('[foo, bar]').toString() === '[ foo, bar ]\n'
```
